### ISSUE TITLE
JavaDoc hover in java editor looks clipped on HiDPI screens

### ISSUE THREAD

**Issue by MikOfClassX** [eclipse-platform/eclipse.platform.swt#1434]
JavaDoc hover in java editor looks clipped on HiDPI screens.

This happens on Win10/Win11, HiDPI monitor, when screen scaling is != 100%

![immagine](https://github.com/user-attachments/assets/74aef160-a23c-4a0c-aa0f-f9b311b5d541)

**Comment by iloveeclipse** [comment]
Could be both HiDPI or Edge related. Are you using edge browser?

Could you please post text info shown if you open Help -> About -> Installation Details -> Configuration and type "SWT" in the filter box?

**Comment by MikOfClassX** [comment]
Id: org.eclipse.e4.ui.css.swt, Version: 0.15.400.v20240321-1245, Location: reference:file:plugins/org.eclipse.e4.ui.css.swt_0.15.400.v20240321-1245.jar
Id: org.eclipse.e4.ui.css.swt.theme, Version: 0.14.400.v20240424-0956, Location: reference:file:plugins/org.eclipse.e4.ui.css.swt.theme_0.14.400.v20240424-0956.jar
Id: org.eclipse.e4.ui.swt.win32, Version: 1.2.300.v20240416-0658, Location: reference:file:plugins/org.eclipse.e4.ui.swt.win32_1.2.300.v20240416-0658.jar
Id: org.eclipse.e4.ui.workbench.addons.swt, Version: 1.5.500.v20240620-1945, Location: reference:file:plugins/org.eclipse.e4.ui.workbench.addons.swt_1.5.500.v20240620-1945.jar
Id: org.eclipse.e4.ui.workbench.renderers.swt, Version: 0.16.500.v20240727-1037, Location: reference:file:plugins/org.eclipse.e4.ui.workbench.renderers.swt_0.16.500.v20240727-1037.jar
Id: org.eclipse.e4.ui.workbench.swt, Version: 0.17.500.v20240807-0911, Location: reference:file:plugins/org.eclipse.e4.ui.workbench.swt_0.17.500.v20240807-0911.jar
Id: org.eclipse.swt, Version: 3.127.0.v20240821-1414, Location: reference:file:plugins/org.eclipse.swt_3.127.0.v20240821-1414.jar
Id: org.eclipse.swt.win32.win32.x86_64, Version: 3.127.0.v20240821-1414, Location: reference:file:plugins/org.eclipse.swt.win32.win32.x86_64_3.127.0.v20240821-1414.jar
Id: org.eclipse.wb.rcp.SWT_AWT, Version: 1.9.500.202403201915, Location: reference:file:plugins/org.eclipse.wb.rcp.SWT_AWT_1.9.500.202403201915.jar
Id: org.eclipse.wb.rcp.swing2swt, Version: 1.9.600.202406171724, Location: reference:file:plugins/org.eclipse.wb.rcp.swing2swt_1.9.600.202406171724.jar
Id: org.eclipse.wb.swt, Version: 1.9.700.202407271418, Location: reference:file:plugins/org.eclipse.wb.swt_1.9.700.202407271418.jar
Id: org.eclipse.wb.swt.widgets.baseline, Version: 1.10.100.202406171724, Location: reference:file:plugins/org.eclipse.wb.swt.widgets.baseline_1.10.100.202406171724.jar

**Comment by iloveeclipse** [comment]
Thanks, you've posted wrong tab content.
Please post Help -> About -> Installation Details -> **Configuration**, section **System properties** filtered by "SWT".

**Comment by MikOfClassX** [comment]
[swt.txt](https://github.com/user-attachments/files/16838627/swt.txt)

**Comment by merks** [comment]
FYI, I tried to reproduce this without success.   I tried different JVMs, and with and without edge.  My resolutions are 125 and 225.

**Comment by MikOfClassX** [comment]
well, i digged some time with dpi-related issues (in swing basically) and my guess is that the popup-webview canvas size does not respect the os screen scaling

**Comment by HeikoKlare** [comment]
> Could be both HiDPI or Edge related. Are you using edge browser?

This must be somehow HiDPI-related, because with Edge currently no local resources (like the image with the icon for a public abstract method) would be displayed. Thus, in the screenshot, IE was used. This is how it would look with Edge:
![49](https://github.com/user-attachments/assets/6bba0468-0cf8-4102-a9e8-8f627ca82c46)
This will be fixed once https://github.com/eclipse-platform/eclipse.platform.swt/pull/1395 and https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/1597 are merged (should be the case soon after master opens again).

**Comment by akoch-yatta** [comment]
@HeikoKlare I found the place and opened a PR for it. Can you have a look?

**Comment by howlger** [comment]
I also see this with the Eclipse SDK 4.33RC2 on Windows 11 too, if and only if the scaling in the Windows settings is set higher than 150%. It affects also the _Javadoc_ view, the _Help > Help Contents_ window, the _Help > Welcome_, etc.

Screenshot with 250% scaling:
![Screenshot 2024-09-02 170601](https://github.com/user-attachments/assets/f5804336-8e63-484b-bee3-5d26d0b12099)

With a scaling of 175% or 250%, the browser widget covers only 1/4 of the X and Y axes (see screenshot above). At 400% it is only 1/16.

By default, the Internet Explorer browser will be used. As a workaround, the Edge browser can be used via the VM argument `-Dorg.eclipse.swt.browser.DefaultType=edge`, but this breaks the _Welcome_ screen (everything else mentioned seems to be fine):

![Screenshot 2024-09-02 172341](https://github.com/user-attachments/assets/f07310ba-1a14-4cf2-bf23-bbebfcde85e9)

**Comment by howlger** [comment]
> @HeikoKlare I found the place and opened a PR for it. Can you have a look?

Wow, that was quick! Could you please share the changed `org/eclipse/swt/ole/win32/OleClientSite.class` file, so I can quickly patch my installation and test this change.

**Comment by HeikoKlare** [comment]
With the fix, everything looks fine for me again:
![image](https://github.com/user-attachments/assets/d1233e01-78ba-419e-8dc2-1bd5bb032030)

Here is a patched `OleClientSite.class` (you have to rename to `.class`, as you cannot upload `.class` files: 
[OleClientSite.txt](https://github.com/user-attachments/files/16839563/OleClientSite.txt)

I am concerned what this means for the upcoming release. This bug obviously affects all OLE-based (including browser-based) views, like welcome, help, Javadoc (view and element information), internal browser etc.
@merks What do you think?

**Comment by howlger** [comment]
> Here is a patched `OleClientSite.class` (you have to rename to `.class`, as you cannot upload `.class` files):
> [OleClientSite.txt](https://github.com/user-attachments/files/16839563/OleClientSite.txt)

Thx. I can confirm that the patched `OleClientSite.class` fixes the issue for me. The hover, _Javadoc_ view, help window and the _Welcome_ screen are displayed as expected, covering the whole area.

**Comment by merks** [comment]
This looks super nasty and is a regression at that.  Fixing it for the release would be very good.  I'm not sure we are capable to producing an RC2a with all the changes taking place to open master for 4.34 development.  I really hope it's possible.  It certainly has my PMC approval.  The change is trivial...

**Comment by akurtakov** [comment]
@MohananRahul how do we stand with building from 4.33 branch ?

**Comment by deepika-u** [comment]
> > Here is a patched `OleClientSite.class` (you have to rename to `.class`, as you cannot upload `.class` files):
> > [OleClientSite.txt](https://github.com/user-attachments/files/16839563/OleClientSite.txt)
> 
> Thx. I can confirm that the patched `OleClientSite.class` fixes the issue for me. The hover, _Javadoc_ view, help window and the _Welcome_ screen are displayed as expected, covering the whole area.

Thanks @howlger
I am also able to recreate the problem with "use internal web browser" as IE and my laptop at 200% resolution as below 
![image](https://github.com/user-attachments/assets/84b77c0a-e20c-47e6-96d6-4e8718eb6a6b)

![image](https://github.com/user-attachments/assets/73c98397-c907-4919-80d1-cecebf1e3e02)


Now when i applied the pr #1435 alone, i am able to see the issue resolved.
![image](https://github.com/user-attachments/assets/5a75cdf5-9a4d-4bc5-8001-1090dc6bb841)

I too have an opinion that this fix should go into 4.33 itself.

**Comment by MohananRahul** [comment]
> @MohananRahul how do we stand with building from 4.33 branch ?

Once the backport fix is done, we can respin. IBuild on R4_33 branch done with [4.33 IBuild moved to R4_33 maintenance branch.](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2277) and  created [PR](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/2298) for publishing.

**Comment by deepika-u** [comment]
With the below build 
Eclipse SDK
Version: 2024-09 (4.33)
Build id: I20240903-0240
OS: Windows 11, v.10.0, x86_64 / win32
Java vendor: Eclipse Adoptium
Java runtime version: 21.0.1+12-LTS
Java version: 21.0.1

I am seeing javadoc issue and welcome screens come properly.

**Comment by deepika-u** [comment]
Hi, When i run the Snippet365, the last set of widgets is missing at 175% resolution. Attaching the screenshots here.

At 125% resolution i see all the 3 widgets displayed(Table, tree, expand bar section).
![image](https://github.com/user-attachments/assets/2e9db2e5-dc96-48c2-a8e0-ff3de209f00b)

But at 175% resolution, i dont see the last set of widgets(Table, tree, expand bar section).
![image](https://github.com/user-attachments/assets/f9530424-be9a-4ddc-9885-a62bb5bbdc0d)

Wrt tables, i tried Snippet349 => this works fine.
wrt tree, i tried Snippet312 => this works fine.
wrt expand bar, i tried Snippet223 => this works fine.

**Comment by merks** [comment]
Note that this issue is specifically for the JavaDoc problem.  Please open an issue for any problem not related to this.

**Comment by MikOfClassX** [comment]
![immagine](https://github.com/user-attachments/assets/5b01b778-a537-470e-b0d0-faa025e48c88)

problem gone in the last version. Good job!

**Comment by akurtakov** [comment]
Closing as fixed.

**Comment by HeikoKlare** [comment]
Note that this currently only fixed on the R4_33_maintenance branch and has to be merged to master as well (via #1435). I didn't want to merge during freeze period but would do it right after master opens. Or do you think we should merge that now to have current release and development branch in sync, @akurtakov? The build of that PR of course complains about the required version bump as the branch is already prepared for 4.34, but conceptually the fix still belongs to 4.33.

**Comment by akurtakov** [comment]
I've rebased on top of master so the build completes. It should have went in master before branch so please merge as soon as build completes.

**Comment by HeikoKlare** [comment]
Yes, in general it should be in master before being backported. But I don't think it's defined how to process such a situation during the release process when the master branch is closed because of being prepared for the next release. In my opinion, we should then merge into master despite freeze as well, but not sure whether that may produce any problems. But that's also a rather seldom problem.
